cupy/cuda/__init__.py

```diff
@@ -69,6 +70,20 @@ def get_cuda_path():
     return _cuda_path
 
 
+@util.memoize()
+def get_attributes_dict(device_id):
```

Comment:
Please move this function to `device.pyx` and name it `_get_attributes` to make it explicitly private.